texml-2.0.2/docs/texml.1
.TH texml 1  "April 16, 2006" "version 2.0"
.SH NAME
texml \- Serialize TeXML to TeX (LaTeX, ConTeXt)
.SH SYNOPSIS
.B texml
[ \fIoptions\fR ] \fIinputfile\fR \fIoutputfile\fR
.SH DESCRIPTION
Convert TeXML file \fIinputfile\fR to TeX file \fIoutfile\fR. The special file name "\fI\-\fR" means the standard input or standard output, correspondingly.
.TP
\fB\-w\fR, \fB\-\-width\fR \fIwidth\fR
If possible, split long lines on smaller ones of the desired width.
.TP
\fB\-e\fR, \fB\-\-encoding\fR \fIencoding\fR
Output encoding. Should be known to the Python codecs. Default is ASCII.
.TP
\fB\-a\fR, \fB\-\-ascii\fR
Use only ASCII symbols for output. Non-ASCII bytes are escaped using the \fB^^XX\fR form, where \fBXX\fR is a hexadecimal code of the character.
.TP
\fB\-c\fR, \fB\-\-context\fR
Activate ConTeXt support.
.SH AUTHOR
Oleg Parashchenko <olpa@users.sourceforge.net>
.SH SEE ALSO
http://getfo.org/texml/

http://uucode.com/blog/category/texml/


texml-2.0.2/ChangeLog
2011-01-19  Oleg Parashchenko  olpa@users.sourceforge.net

* Texml/handler.py: wrapper around strip() to support very
old python version
* scripts/texml.py: version number is changed to 2.0.2.

2006-07-20  Oleg Parashchenko  olpa@users.sourceforge.net

* Texml/texml.py: try to write a symbol in the ^^XX form first,
and only then consult the rewriting maps. Otherwise combination
of --encoding and --ascii gives undesired result.
* scripts/texml.py: version number is changed to 2.0.1.


2006-07-10  Oleg Parashchenko  olpa@users.sourceforge.net

* TeXML 2.0.0 release.


2006-06-14  Oleg Parashchenko  olpa@users.sourceforge.net

* Texml/handler.py: one raise was incorrectly commented-out.
* scripts/texml.py: version number is changed to 2.0.0.beta1.
* scripts/texml.py: the first line now is just "#!python", the
path is to be corrected by distutils.


2006-06-02  Oleg Parashchenko  olpa@users.sourceforge.net

* Texml/specmap.py: mapping for the non-breaking hyphen,
as suggested by Paul Tremblay.


2006-06-01  Oleg Parashchenko  olpa@users.sourceforge.net

* Texml/specmap.py: "|" is now "\textbar".


2005-10-03  Oleg Parashchenko  olpa@users.sourceforge.net

* scripts/texml/texml.py: First line now is "#!python".
* scripts/texml: New file.
* scripts/texml_local: Old content of PYTHONPATH is retained.
* tests/data/quick/Makefile: Switched from using "texml.py" to
using "texml_local".
* setup.py: Version string is extracted from "scripts/texml.py".
* MANIFEST.in: HTML and text documentation is added.
* README: typo is fixed.


2005-09-26  Oleg Parashchenko  olpa@users.sourceforge.net

* texml.py: Version number is updated: 1.29.devel.
* texml.py: Default width now is 62 symbols.
* texmlwr.py: If symbol couldn't be translated to an encoding,
  command "\unicodechar" is written instead of an #&xNNN; entity.


2005-09-07  Oleg Parashchenko  olpa@users.sourceforge.net

* texml.py: Version number is updated: 1.28.devel.
* dtd/texml.dtd: Element "pdf" is added.
* handler.py, texmlwr.py: Support of the element "pdf".
* data/pdfstring.xml, data/pdfstring.out, faildata/pdfstring.xml,
  faildata/pdfstring.grep, test/readme: Tests for the "pdf" element.
* handler.py: doesn't print error on it's own. Instead, returns the
  text to the caller, and caller should print the message.
  The function "__print_error" is deleted. The format "prefix:\n"
  is changed to "prefix: ".
* unimap.py: no more "&lt;" and "&gt;" instead of "<" and ">".
  Thanks Felix Wiemann for reporting the bug.
  http://sourceforge.net/mailarchive/forum.php?thread_id=7467879&forum_id=44318


Version 1.27.devel (20 June 2005)

* New feature: "--ascii" parameter. Now it's possible to
  generate plain ASCII TeX files in a desired encoding.
  Non-ascii bytes are encoded as "^^XX".
  The test file "chinese1.xml" is a working example of
  a Chinese LaTeX file.

* TeXML issues a warning is an XML symbol isn't converted to
  TeX and printed as '&#xNNN;'

* Refactoring. Code for tuning output stream is moved from
  "handler.py" to "texmlwr.py".


Version 1.26.devel (8 May 2005)

* Bug is fixed. Now lines which consist of space characters and
  tabulation characters are also regarded as blank lines.


Version 1.25.devel (4 April 2005)

* Script texml_con
  This script can directly convert a texml 
  document to a print document such as PDF.

* External entities now handled
  A small bug in handler.py prevented external 
  entties from being read. External entities
  can now be read.


Version 1.24.devel (21 March 2005)

* TeXML is now namespace aware
  will process documents with the namespace 
  http://getfo.sourceforge.net/texml/ns1

  will also process documents with no namespace

* TeXML is installed the standard way
  libraries are created in the python path;
  scripts are automatically installed in the path

* Error handling
  
  exceptions are now classes rather than strings. 
  This allows for better handling of exceptions.

  Column and line numbers are now outputed as part 
  of the error messages when an exception is raised.

  exceptions for invalid or ill-formed XML are now
  handled in the script. Instead of a long message
  (incomprehensible to non-python developers), the
  error message consists of a column number, a line
  number, and a simple explanation. 
  

Version 1.23.devel (5 March 2005)

* ConTeXt mode is now enabled through command line instead of
  using the environment variable.


Version 1.22.devel (21 February 2005)

* Support of ConTeXt environments (\startenv ... \stopenv)
  Thanks to Paul Tremblay for the patch. More details:
  http://sourceforge.net/tracker/index.php?func=detail&aid=1145206&group_id=102261&atid=631462
  http://sourceforge.net/forum/message.php?msg_id=3007157


Version 1.21.devel (18 February 2005)

* TeXML now should work under Python 2.2.1.
  Thanks to Paul Tremblay for the bug report and the fix suggestion:
  http://sourceforge.net/tracker/index.php?func=detail&aid=1123627&group_id=102261&atid=631460


Version 1.20 (12 November 2004)

* Jason Santos <jason.santos@megaslow.net> has proofread the documentation.
* Corrected DTD: attribute spec/@cat now allows the value "nl?".


Version 1.15.devel (17 August 2004)

* New category of special items: conditional newline.


Version 1.12.devel (09 August 2004)

* New document: Thesis of the talk at the First International
  Conference of Open-Source Developers.
* Names of distribution packages now contain suffixes '-windows'
  and '-unix' to reflect a difference in line endings.


Version 1.09.devel (08 July 2004)

* Environments and groups are now allowed in parameters.
* Whitespace processing in environments is reworked. Now @nl2
  attribute of <env/> regards possible <opt/> and <parm/>.
* Empty line, produced by <spec cat="nl"/>, is no longer being
  commented out.


Version 1.08.devel (29 June 2004)

* Advanced whitespace processing now implemented: the program
  (a) removes what XML regards as insignificant whitespace, and

texml-2.0.2/scripts/texml.py
#!python
# $Id: texml.py,v 1.20 2006-07-20 03:56:49 olpa Exp $

VERSION = "2.0.2"; # GREPVERSION # Format of this string is important
usage = """Convert TeXML markup to [La]TeX markup. v.%s. Usage:
python texml.py [-e encoding] [-w auto_width] [-c|--context] [-a|--ascii] in_file out_file""" % VERSION

#
# Check command line, print help
#
import sys
if len(sys.argv) < 3:
  print >>sys.stderr, usage
  sys.exit(1)

#
# Parse command line options
#
encoding      = 'ascii'
always_ascii  = 0
width         = 62
use_context   = 0
use_namespace = 1
import getopt
try:
  opts, args = getopt.getopt(sys.argv[1:], 'hcaw:e:', ['help', 'context', 'ascii', 'width=', 'encoding=', ])
except getopt.GetoptError, e:
  print >>sys.stderr, 'texml: Can\'t parse command line: %s' % e
  print >>sys.stderr, usage
  sys.exit(2)
for o, a in opts:
  if o in ('-h', '--help'):
    print >>sys.stderr, usage;
    sys.exit(1)
  if o in ('-c', '--context'):
    use_context = 1
  if o in ('-a', '--ascii'):
    always_ascii = 1
  if o in ('-w', '--width'):
    try:
      width = int(a)
    except:
      print >>sys.stderr, "texml: Width is not an integer: '%s'" % a
      sys.exit(1)
    if width < 3: # just a random value
      print >>sys.stderr, "texml: Width should be greater 3 but get '%s'" % a
      sys.exit(1)
  if o in ('-e', '--encoding'):
    encoding = a

#
# Get input and output file
#
if len(args) != 2:
  print >>sys.stderr, 'texml: Expected two command line arguments, but got %d' % len(args)
  sys.exit(3)
(infile, outfile) = args

#
# Prepare transformation-1: input file, XML parser
#
import xml.sax 
from xml.sax.handler import feature_namespaces

if '-' == infile:
  infile = sys.stdin

#
# Prepare transformation-2: output
#
if '-' == outfile:
  f = sys.stdout
else:
  f = file(outfile, 'wb')

#
# An error handler
#
def quit(msg):
    sys.stderr.write(msg)
    f.close()
    sys.exit(1)

#
# import main class and parse
#
import Texml.processor
try:
  Texml.processor.process(infile, f, encoding, width, always_ascii, use_context)
    
except Exception, msg:
  msg = 'texml: %s\n' % (str(msg))
  quit(msg)

f.close()
sys.exit(0)
